core/src/main/scala/quasar/destination/gbq/GBQAccessToken.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.gbq
+
+import cats.effect._
+import cats.effect.{Blocker, Sync}
+
+import com.google.auth.oauth2.AccessToken
+import com.google.auth.oauth2.GoogleCredentials
+
+import java.io.ByteArrayInputStream
+import java.util.concurrent.Executors
+
+import quasar.concurrent.NamedDaemonThreadFactory
+
+import scala.{
+  Array,
+  Byte
+}
+import scala.concurrent.ExecutionContext
+
+object GBQAccessToken {
+  private def genAccessToken[F[_]: Sync](auth: Array[Byte]): F[AccessToken] = Sync[F] delay {
+    val authInputStream = new ByteArrayInputStream(auth)
+    val credentials = GoogleCredentials
+      .fromStream(authInputStream)
+      .createScoped("https://www.googleapis.com/auth/bigquery")
+    credentials.refreshIfExpired
+    credentials.refreshAccessToken
```

Comment:
@cryogenian, if I use `Sync[F].dealy` like you mention, should remove the outer `Sync[F].dealy` wrapping the entire function definition?

Reply:
```scala
  private def genAccessToken[F[_]: Sync](auth: Array[Byte]): F[AccessToken] = {
    val credentials = Sync[F] delay {
      val authInputStream = new ByteArrayInputStream(auth)
      GoogleCredentials
        .fromStream(authInputStream)
        .createScoped("https://www.googleapis.com/auth/bigquery")
    }
    credentials.flatMap(creds =>
      Sync[F].delay(creds.refreshIfExpired()) >>
        Sync[F].delay(creds.refreshAccessToken())
  }
```
Here's one way this could be structured. This makes it more clear that we're throwing away the value of `credentials.refreshIfExpired()` and returning the value of `credentials.refreshAccessToken()`.